mk/artifact.mk

```diff
@@ -48,12 +48,27 @@ else
   LDFLAGS_CROSS := -lm -lsemihost
 endif
 
-.PHONY: artifact scimark2 ieeelib
+.PHONY: artifact fetch-sha1 scimark2 ieeelib
 
-artifact: ieeelib scimark2
+artifact: fetch-sha1 ieeelib scimark2
 ifeq ($(call has, PREBUILT), 1)
-	$(Q)$(PRINTF) "Fetching prebuilt executables from \"rv32emu-prebuilt\" ...\n"
-	$(Q)wget -q --show-progress https://github.com/sysprog21/rv32emu-prebuilt/releases/download/$(LATEST_RELEASE)/rv32emu-prebuilt.tar.gz -O- | tar -C build --strip-components=1 -xz
+	$(Q)$(PRINTF) "Checking SHA1 of binaries ...\n"
+
+	$(eval PREBUILT_X86_FILENAME := $(shell cat $(BIN_DIR)/sha1sum-linux-x86-softfp | awk '{ for(i=1;i<=NF;i++) if(i%2==0) print $$i };'))
+	$(Q)$(eval PREBUILT_RV32_FILENAME := $(shell cat $(BIN_DIR)/sha1sum-riscv32 | awk '{ for(i=1;i<=NF;i++) if(i%2==0) print $$i };'))
+
+	$(Q)$(eval RES := 0)
+	$(Q)$(eval $(foreach FILE,$(PREBUILT_X86_FILENAME), \
+	    $(call verify,$(shell grep -w $(FILE) $(BIN_DIR)/sha1sum-linux-x86-softfp | awk '{ print $$1 };'),$(BIN_DIR)/linux-x86-softfp/$(FILE),RES) \
+	))
+	$(Q)$(eval $(foreach FILE,$(PREBUILT_RV32_FILENAME), \
+	    $(call verify,$(shell grep -w $(FILE) $(BIN_DIR)/sha1sum-riscv32 | awk '{ print $$1 };'),$(BIN_DIR)/riscv32/$(FILE),RES) \
+	))
+
+	$(Q)if [ "$(RES)" = "1" ]; then \
+	    $(PRINTF) "SHA1 verification failed! Re-fetching prebuilt binaries from \"rv32emu-prebuilt\" ...\n"; \
```

Comment:
Add some color to this message, such as using red.

define red color in `mk/common.mk`
```
RED = \033[31m
```

Consider this: `$(PRINTF) "$(RED)SHA1 verification failed! Re-fetching prebuilt binaries from \"rv32emu-prebuilt\" ...\n$(NO_COLOR)"`